crates/router/src/core/payments/flows/authorize_flow.rs

```diff
@@ -200,10 +203,90 @@ impl Feature<api::Authorize, types::PaymentsAuthorizeData> for types::PaymentsAu
                 &new_router_data.request,
                 &new_router_data.response,
             );
-
             new_router_data.integrity_check = integrity_result;
-
-            metrics::PAYMENT_COUNT.add(&metrics::CONTEXT, 1, &[]); // Metrics
+            // Check if the Capture API should be called based on the connector and other parameters
+            if helpers::should_call_capture_api(
+                &connector.connector_name,
+                self.request.off_session,
+                self.request.customer_acceptance,
+                self.request.capture_method,
+                self.request.setup_future_usage,
+                new_router_data.status,
+            ) {
+                // Convert Authorize RouterData into Capture RouterData
+                let mut capture_router_data = helpers::router_data_type_conversion(
```

Comment:
make separate two seperate, one till decide_flow and another one for handling response.

Reply:
added
two seperate functions in the flows.rs